onnxruntime/python/tools/transformers/models/stable_diffusion/diffusion_models.py

```diff
@@ -0,0 +1,855 @@
+
+        self.graph = gs.import_onnx(onnx_graph)
+
+
+class PipelineInfo:
+    def __init__(
+        self, version: str, is_inpaint: bool = False, is_sd_xl_refiner: bool = False, use_vae_in_xl_base=False
+    ):
+        self.version = version
+        self._is_inpaint = is_inpaint
+        self._is_sd_xl_refiner = is_sd_xl_refiner
+        self._use_vae_in_xl_base = use_vae_in_xl_base
+
+        if is_sd_xl_refiner:
+            assert self.is_sd_xl()
+
+    def is_inpaint(self) -> bool:
+        return self._is_inpaint
+
+    def is_sd_xl(self) -> bool:
+        return "xl" in self.version
+
+    def is_sd_xl_base(self) -> bool:
+        return self.is_sd_xl() and not self._is_sd_xl_refiner
+
+    def is_sd_xl_refiner(self) -> bool:
+        return self.is_sd_xl() and self._is_sd_xl_refiner
+
+    def use_safetensors(self) -> bool:
+        return self.is_sd_xl()
+
+    def stages(self) -> List[str]:
+        if self.is_sd_xl_base():
+            return ["clip", "clip2", "unetxl"] + (["vae"] if self._use_vae_in_xl_base else [])
+
+        if self.is_sd_xl_refiner():
+            return ["clip2", "unetxl", "vae"]
+
+        return ["clip", "unet", "vae"]
+
+    def vae_scaling_factor(self) -> float:
+        return 0.13025 if self.is_sd_xl() else 0.18215
+
+    @staticmethod
+    def supported_versions():
+        return ["1.4", "1.5", "2.0-base", "2.0", "2.1", "2.1-base", "xl-1.0"]
+
+    def name(self) -> str:
+        if self.version == "1.4":
+            if self.is_inpaint():
+                return "runwayml/stable-diffusion-inpainting"
+            else:
+                return "CompVis/stable-diffusion-v1-4"
+        elif self.version == "1.5":
+            if self.is_inpaint():
+                return "runwayml/stable-diffusion-inpainting"
+            else:
+                return "runwayml/stable-diffusion-v1-5"
+        elif self.version == "2.0-base":
+            if self.is_inpaint():
+                return "stabilityai/stable-diffusion-2-inpainting"
+            else:
+                return "stabilityai/stable-diffusion-2-base"
+        elif self.version == "2.0":
+            if self.is_inpaint():
+                return "stabilityai/stable-diffusion-2-inpainting"
+            else:
+                return "stabilityai/stable-diffusion-2"
+        elif self.version == "2.1":
+            return "stabilityai/stable-diffusion-2-1"
+        elif self.version == "2.1-base":
+            return "stabilityai/stable-diffusion-2-1-base"
+        elif self.version == "xl-1.0":
+            if self.is_sd_xl_refiner():
+                return "stabilityai/stable-diffusion-xl-refiner-1.0"
+            else:
+                return "stabilityai/stable-diffusion-xl-base-1.0"
+
+        raise ValueError(f"Incorrect version {self.version}")
+
+    def short_name(self) -> str:
+        return self.name().split("/")[-1].replace("stable-diffusion", "sd")
+
+    def clip_embedding_dim(self):
+        # TODO: can we read from config instead
```

Comment:
Can we read from the config?

Reply:
Right now the code only deal with a set of fixed models, so it is fine right now. Will consider it when we need support any model in the future.